storage/objects/stream_file_upload.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package objects
+
+// [START storage_stream_file_upload]
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"cloud.google.com/go/storage"
+)
+
+// streamFileUpload uploads an object via a stream.
+func streamFileUpload(w io.Writer, bucket, object string) error {
+	// bucket := "bucket-name"
+	// object := "object-name"
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("storage.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	b := []byte("Hello world.")
+	buf := bytes.NewBuffer(b)
+
+	ctx, cancel := context.WithTimeout(ctx, time.Second*50)
+	defer cancel()
+
+	// Upload an object with storage.Writer.
+	wc := client.Bucket(bucket).Object(object).NewWriter(ctx)
+	wc.ChunkSize = 0 // note retries are not supported for chunk size 0
+	if _, err = io.Copy(wc, buf); err != nil {
+		return fmt.Errorf("io.Copy: %v", err)
+	}
+	// Data can continue to be added to the file until the writer is closed
```

Comment:
Should we add an example of this? Perhaps copy in another buffer?

also nit: comment should end with a period to be consistent with the rest of the samples

Reply:
Spoke with @tritone regarding this and we decided that it might over-complicate the sample to add an example. Go users would presumably know how to do this since it is not a storage operation. What do you think? I'm okay either way.

Reply:
Yeah, that's true. The comment should probably be enough